pyproject.toml

```diff
@@ -15,9 +15,9 @@ classifiers = [
 python = ">=3.9.0,<3.11"
 numpy = "^1.24"
 pandas = "^2.0"
-flwr = "1.4.0"
+flwr = "1.7.0"
 opacus = "^1.3.0"
-torch = "^1.12.1"
+torch = ">=2.0.0, !=2.0.1, !=2.1.0"
```

Comment:
This is risky, if torch 3 is launched it can get upgraded by accident. Also, those exceptions seem pretty complex. How about not the following:
```toml
torch = "^2.1.1"
```

Reply:
I originally tried something similar but due to a bug in poetry, I had to resort to a rather convoluted constraint: https://stackoverflow.com/questions/76327419/valueerror-libcublas-so-0-9-not-found-in-the-system-path

But on second glance, upgrading to 2.1.1 or higher should be fine. So I will try what you suggested

Reply:
Huh that's a curious problem, I've never seen it before.